File: LeetCode/data_structure/p217_contains_duplicate/c++/DuplicateFinder.cpp
#include<iostream>
#include<vector>
#include<map>
using namespace std;

bool containsDuplicate(vector<int>& nums) {
    map<int, int> numsContainer;
    for(auto & num: nums) {
        if(numsContainer.find(num)->second) {
            return true;
        }
        numsContainer.insert(pair<int, int> (num, 1));
    }
    return false;
}

int main() {
    vector<int> nums{1,2,3,1};
    cout << containsDuplicate(nums);
    return 0;
}

File: LeetCode/data_structure/p217_contains_duplicate/python/DuplicateFinder.py
def main():
    print("Hello World!")

def contains_duplicate(nums: list) -> bool:
    numsContainer = {}
    for num in nums:
        if num in numsContainer:
            return True
        numsContainer[num] = 1
    return False


if __name__ == "__main__":
    print(contains_duplicate([1,2,3,4])) # False
    print(contains_duplicate([1,2,3,1])) # True